attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto

```diff
@@ -137,3 +138,39 @@ message AttributeMetadataFilter {
 message Empty {
 }
 
+message AttributeDefinition {
+  oneof value {
+    string span_path = 1;
+    Projection projection = 2;
+  }
+}
+
+message Projection {
+  oneof value {
+    string attribute_key = 1;
```

Comment:
Is this like a foreign key into another attribute? When would we do that?

Reply:
It is. The simple case here would be something like an alias. But this is also used in how we recursively define something in terms of other attributes. Take the definition of an attribute like serviceId, assuming for sake of discussion we want it to be based solely on service name:
```
 definition = {
   expression = {
     operator = HASH
     arguments = [{
       attribute_key = "serviceName"
     }]
   }
 }
```